js/src/util/validation.spec.js

```diff
@@ -0,0 +1,303 @@
+      expect(validateName(null)).to.deep.equal({
+        name: null,
+        nameError: ERRORS.invalidName
+      });
+    });
+
+    it('fails on short names', () => {
+      expect(validateName('  1  ')).to.deep.equal({
+        name: '  1  ',
+        nameError: ERRORS.invalidName
+      });
+    });
+  });
+
+  describe('validatePositiveNumber', () => {
+    it('validates numbers', () => {
+      expect(validatePositiveNumber(123)).to.deep.equal({
+        number: 123,
+        numberError: null
+      });
+    });
+
+    it('validates strings', () => {
+      expect(validatePositiveNumber('123')).to.deep.equal({
+        number: '123',
+        numberError: null
+      });
+    });
+
+    it('validates bignumbers', () => {
+      expect(validatePositiveNumber(new BigNumber(123))).to.deep.equal({
+        number: new BigNumber(123),
+        numberError: null
+      });
+    });
+
+    it('fails on invalid numbers', () => {
+      expect(validatePositiveNumber(null)).to.deep.equal({
+        number: null,
+        numberError: ERRORS.invalidAmount
+      });
+    });
+
+    it('fails on negative numbers', () => {
+      expect(validatePositiveNumber(-1)).to.deep.equal({
+        number: -1,
+        numberError: ERRORS.invalidAmount
+      });
+    });
+  });
+
+  describe('validateUint', () => {
+    it('validates numbers', () => {
+      expect(validateUint(123)).to.deep.equal({
+        value: 123,
+        valueError: null
+      });
+    });
+
+    it('validates strings', () => {
+      expect(validateUint('123')).to.deep.equal({
+        value: '123',
+        valueError: null
+      });
+    });
+
+    it('validates bignumbers', () => {
+      expect(validateUint(new BigNumber(123))).to.deep.equal({
+        value: new BigNumber(123),
+        valueError: null
+      });
+    });
+
+    it('fails on invalid numbers', () => {
+      expect(validateUint(null)).to.deep.equal({
+        value: null,
+        valueError: ERRORS.invalidNumber
+      });
+    });
+
+    it('fails on negative numbers', () => {
+      expect(validateUint(-1)).to.deep.equal({
+        value: -1,
+        valueError: ERRORS.negativeNumber
+      });
+    });
+
+    it('fails on decimal numbers', () => {
+      expect(validateUint(3.1415927)).to.deep.equal({
+        value: 3.1415927,
+        valueError: ERRORS.decimalNumber
+      });
+    });
+  });
+
+  describe('isNullAddress', () => {
+    it('verifies a prefixed null address', () => {
+      expect(isNullAddress(`0x${NULL_ADDRESS}`)).to.be.true;
+    });
+
+    it('verifies an non-prefixed null address', () => {
+      expect(isNullAddress(NULL_ADDRESS)).to.be.true;
+    });
+
+    it('fails on a null value', () => {
+      expect(isNullAddress(null)).to.be.false;
+    });
+
+    it('fails on a non-full length 00..00 value', () => {
+      expect(isNullAddress(NULL_ADDRESS.slice(-1 * (NULL_ADDRESS.length - 2)))).to.be.false;
+    });
+
+    it('fails on a valid addess, non 00..00 value', () => {
```

Comment:
[tiny grumble] "fails" indicates that it throws an error, but it rather returns false.